src/message-store-sql.ts

```diff
@@ -197,22 +197,18 @@ export class MessageStoreSql implements MessageStore {
     query = filterSelectQuery(filters, query);
 
     // extract sort property and direction from the supplied messageSort
-    const { property: sortProperty, direction: sortDirection } = this.getOrderBy(messageSort);
+    const { property: sortProperty, direction: sortDirection } = this.extractSortProperties(messageSort);
 
     if(pagination?.cursor !== undefined) {
-      const messageCid = pagination.cursor;
-      query = query.where(({ eb, selectFrom, refTuple }) => {
-        const direction = sortDirection === SortDirection.Ascending ? '>' : '<';
-
-        // fetches the cursor as a sort property tuple from the database based on the messageCid.
-        const cursor = selectFrom('messageStore')
-          .select([sortProperty, 'messageCid'])
-          .where('tenant', '=', tenant)
-          .where('messageCid', '=', messageCid)
-          .limit(1).$asTuple(sortProperty, 'messageCid');
+      // currently the sort property is explicitly either `dateCreated` | `messageTimestamp` | `datePublished` which are all strings
+      // if it is not of type string we return an empty result set
+      const cursorValue = pagination.cursor.value as string;
```

Comment:
Just wanted to double check if this comment is still valid. I'm not immediately seeing an empty result set return if `values` isn't a string here.